yarn-project/ethereum/src/utils.ts

```diff
@@ -80,7 +92,43 @@ export function prettyLogViemErrorMsg(err: any) {
   return err?.message ?? err;
 }
 
-export function formatViemError(error: any): string {
+/**
+ * Formats a Viem error into a FormattedViemError instance.
+ * @param error - The error to format.
+ * @param abi - The ABI to use for decoding.
+ * @returns A FormattedViemError instance.
+ */
+export function formatViemError(error: any, abi: Abi = RollupAbi): FormattedViemError {
```

Comment:
IIRC Lasse had set up a combined abi with all errors from all contracts. We should use that here.

Reply:
I altered the script that we use to generate the artifacts such that the errors should be included inside the other ABIs. So they are in the rollup one or whetever is being used. But maybe it should not be specifically the `RollupAbi` that is being checked against here. Mainly if some other contract were used this formatter gets pretty strange.

https://github.com/AztecProtocol/aztec-packages/pull/10697/files

Reply:
hmmm so I just added an `ErrorsAbi` in [this commit](https://github.com/AztecProtocol/aztec-packages/pull/11211/commits/d51c6995ec3537e61db95643d46bfbfb3732b0d2) but happy to rollback. Not sure what the issue is you've mentioned in this comment, we'd only change the output in `l1-artifacts` typescript project.

This function should be able to work with any ABI as input, was just using `RollupAbi` as default since that's where we're using it for now in l1-publisher